feat: add more options to configure diff output (#2522)
vitest-dev/vitest: docs/config/index.md

### outputTruncateLength

- **Type:** `number`
- **Default:** `stdout.columns || 80`
- **CLI:** `--outputTruncateLength <length>`, `--output-truncate-length <length>`

Truncate the size of diff line up to `stdout.columns` or `80` number of characters. You may wish to tune this, depending on your terminal window width. Vitest includes `+-` characters and spaces for this. For example, you might see this diff, if you set this to `6`:

```diff
// actual line: "Text that seems correct"
- Text...
+ Test...
```

### outputDiffLines

- **Type:** `number`
- **Default:** `15`
- **CLI:** `--outputDiffLines <lines>`, `--output-diff-lines <lines>`

> Limit number of single output diff lines up to `15`. Vitest counts all `+-` lines, when determining when to stop. For example, you might see diff like this, if you set this property to `3`:

Reviewer: ```suggestion
Limit the number of single output diff lines up to `15`. Vitest counts all `+-` lines when determining when to stop. For example, you might see diff like this, if you set this property to `3`:
```

```diff
- test: 1,
+ test: 2,
- obj: '1',
...
- test2: 1,
+ test2: 1,
- obj2: '2',
...
```

### outputDiffMaxLines

- **Type:** `number`
- **Default:** `50`
- **CLI:** `--outputDiffMaxLines <lines>`, `--output-diff-max-lines <lines>`
- **Version:** Since Vitest 0.26.0

> Maximum number of lines to display in diff window. Beware that if you have a large object with many small diffs, you might not see all of them at once.

Reviewer: ```suggestion
The maximum number of lines to display in diff windows. Beware that if you have a large object with many small diffs, you might not see all of them at once.
```

### outputDiffMaxSize

- **Type:** `number`
- **Default:** `10000`
- **CLI:** `--outputDiffMaxSize <length>`, `--output-diff-max-size <length>`
- **Version:** Since Vitest 0.26.0

> Maximum length of stringified object before diff happens. Vitest tries to stringify an object before doing a diff, but if the object is too large, it will reduce the depth of the object to fit within this limit. Because of this if object is too big or nested, you might not see the diff.

Reviewer: ```suggestion
The maximum length of the stringified object before the diff happens. Vitest tries to stringify an object before doing a diff, but if the object is too large, it will reduce the depth of the object to fit within this limit. Because of this, if the object is too big or nested, you might not see the diff.
```
